### PR TITLE
GPU Multiple process support

### PR DESCRIPTION
**Pull request by yashanand1910**
## Describe your changes
This PR caters to the `cedana-gpu` PR for adding multiple process support. Please merge it with: 

Now, you should be able to call `cedana exec --gpu-enabled` to launch multiple GPU processes with the daemon and perform C/R.

## Issue ticket number
CED-748

## Checklist before requesting a review
- [x] Have I performed a self-review?
- [x] Have I added regression or unit tests (where it makes sense) for my changes?
- [x] Have I updated the README if changes have resulted in it being out of date?
- [x] Have I checked to ensure my PR only introduces the changes it's intended to? E.g. No extraneous formatting changes.
- [ ] Have I ensured that there are no performance regressions by checking benchmark results?
- [ ] Is this a breaking change? If yes, please describe areas affected and request reviews from developer stakeholders.

### CODE REVIEW

**Review by swarnimarun — COMMENTED** [review]
looks fine to me, just a couple things.

File: pkg/api/dump.go [hunk 1]
```diff
@@ -491,32 +492,24 @@ func (s *service) kataDump(ctx context.Context, state *task.ProcessState, args *
...
-	var opts []grpc.DialOption
-	opts = append(opts, grpc.WithTransportCredentials(insecure.NewCredentials()))
 
-	gpuConn, err := grpc.Dial("127.0.0.1:50051", opts...)
-	if err != nil {
-		log.Warn().Msgf("could not connect to gpu controller service: %v", err)
-		return err
+	gpuController := s.GetGPUController(jid)
+	if gpuController == nil {
+		return fmt.Errorf("did not find gpu controller for job %s", jid)
```
[the remark this answers is not in the document]
Comment (yashanand1910) [reply pkg/api/dump.go:504]: Errors just thrown are eventually caught by the gRPC logging interceptor.

File: pkg/api/dump.go [hunk 2]
```diff
@@ -356,13 +354,16 @@ func (s *service) dump(ctx context.Context, state *task.ProcessState, args *task
 		return err
 	}
 
+	log.Debug().Interface("state", state).Msg("beginning dump")
```
[the remark this answers is not in the document]
Comment (yashanand1910) [reply pkg/api/dump.go:357]: Good catch, this was for debugging. Removing.

File: pkg/api/gpu.go [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+	}
+	defer func() {
+		err = s.StopGPUController("healthcheck")
+		if err != nil {
+			log.Error().Err(err).Msg("could not stop gpu controller")
+		}
+		err = s.WaitGPUController("healthcheck")
+		if err != nil {
+			log.Error().Err(err).Msg("could not wait for gpu controller")
+		}
```
[the remark this answers is not in the document]
Comment (yashanand1910) [reply pkg/api/gpu.go:86]: Stop kills, and wait just collects. Stop will not fail if killing as the daemon (`sudo`), unless the process already exited - in which case, wait will succeed and collect exit code.

**Review by swarnimarun — APPROVED** [review]
lgtm

**Pull request merged**